### PR TITLE
Amend to fix #2474

### LINKED ISSUE #2474: "soft" _isResetOnRevisit breaks questions
Questions set to use `"_isResetOnRevisit": "soft"` within the components.json, cannot be answered. The `_buttonState` is undefined and causes an error, and the question doesn't think its interactive, so items can't be selected.

### PR DESCRIPTION
**Pull request by danielghost**
QuestionModel.reset was never being called.

`_userAnswer` isn't reset via `resetQuestionOnRevisit` but that doesn't directly affect this particular issue.

### CODE REVIEW

**Comment by panthimanshu79** [comment]
It indeed solve my problem too

**Comment by panthimanshu79** [comment]
+1

File: src/core/js/models/itemsQuestionModel.js [hunk 1]
```diff
@@ -159,6 +159,11 @@ define([
             this.set('_isAtLeastOneCorrectSelection', false);
         },
 
+        reset: function(type, force) {
+            QuestionModel.prototype.reset.call(this, type, force);
```
Comment (oliverfoster) [inline src/core/js/models/itemsQuestionModel.js]: `.apply(this, arguments);` is a bit more future-safe as it forwards all supplied arguments without bias.

Comment (moloko) [inline src/core/js/models/itemsQuestionModel.js]:
```suggestion
            QuestionModel.prototype.reset.apply(this, arguments);
```

File: src/core/js/models/itemsQuestionModel.js [hunk 2]
```diff
@@ -159,6 +159,11 @@ define([
             this.set('_isAtLeastOneCorrectSelection', false);
         },
 
+        reset: function(type, force) {
+            QuestionModel.prototype.reset.apply(this, arguments);
+            ItemsComponentModel.prototype.reset.call(this, type, force);
```
Comment (moloko) [inline src/core/js/models/itemsQuestionModel.js]:
```suggestion
            ItemsComponentModel.prototype.reset.apply(this, arguments);
```

**Pull request merged**